modules/minifact/tasks/ruby.rb

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+def facter_executable
+  if ENV.key? 'Path'
+    ENV['Path'].split(';').each do |p|
+      if p =~ /Puppet\\bin\\?$/
+        return File.join(p, 'facter')
+      end
+    end
+    'C:\Program Files\Puppet Labs\Puppet\bin\facter'
```

Comment:
I want to double check this works, but this all makes sense to me.

Reply:
Works great 👍

Reply:
It does at least on the version of Windows I'm testing it on:
```
C:\Users\Administrator>"C:\Program Files\Puppet Labs\Puppet\sys\ruby\bin\ruby" -e "exec('C:\Program Files\Puppet Labs\Puppet\bin\facter', '--json', 'os')"
{
  "os": {
    "architecture": "x64",
    "family": "windows",
    "hardware": "x86_64",
    "name": "windows",
    "release": {
      "full": "2012 R2",
      "major": "2012 R2"
    },
    "windows": {
      "system32": "C:\\Windows\\system32"
    }
  }
}
```